### DISCUSSION TITLE
using keystone as a headless backend

### DISCUSSION THREAD

**Discussion by ahmed-adly-khalil** [keystonejs/keystone#6599] [Event Q&A Q3 2021]
Hello, 
I'm trying to use Keystone as an `admin` UI and use the graphql api and an `application API` layer and build my own application UI using Nuxt, in this context I have used https://auth.nuxtjs.org/ to talk to Keystone authentication API and things are working on term of athentication which happens as follow 

```  mutation authenticateUserWithPassword($email: String!, $password: String!) {
    authenticateUserWithPassword(email: $email, password: $password) {
      __typename
      ... on UserAuthenticationWithPasswordSuccess {
        item {
          name
          email
          id
        }
        sessionToken
        
      }
      ... on UserAuthenticationWithPasswordFailure {
        code
        message
      }
    }
  }
```

the issue now comes when I try to use the `sessionToken` after login to retrieve some data, for ex in the below query i want to pass the sessionToken instead of user ID:

```query customerInfoById($id:ID!){
  customerInfoById:allUsers(where:{id:$id}){
    id
    name
    surname
    email

    profile_picture{
      publicUrl
    }
  }
}
``` 
is there a way to achieve this?

**Comment by bladey** [comment]
Hey @ahmed-adly-khalil,

Thanks for your question! It looks like you're stuck making progress with your project.

To get the most out of the live event, we're keeping this Community Q&A category focused on questions that are relevant to the community as a whole.

Seeing as this question is specific to your project, would you mind posting it to our community Slack, that's the best place to get focused support on blocks you encounter with Keystone - https://community.keystonejs.com

**Reply by ahmed-adly-khalil** [reply]
awesome, done: https://keystonejs.slack.com/archives/C01STDMEW3S/p1632145496303000
